src/tiled/scriptdialog.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * scriptdialog.h
+ * Copyright 2020, David Konsumer <konsumer@jetboystudio.com>
+ * Copyright 2020, Thorbjørn Lindeijer <bjorn@lindeijer.nl>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "mainwindow.h"
+#include "fileedit.h"
+#include "colorbutton.h"
+#include <QString>
+#include <QObject>
+#include <QtWidgets/QDialogButtonBox>
+#include <QtWidgets/QLabel>
+#include <QtWidgets/QHBoxLayout>
+#include <QtWidgets/QVBoxLayout>
+#include <QtWidgets/QWidget>
+#include <QtWidgets/QDoubleSpinBox>
+#include <QtWidgets/QSlider>
+#include <QtWidgets/QCheckBox>
+#include <QtWidgets/QPushButton>
+#include <QComboBox>
+#include <QCoreApplication>
+#include <QDialog>
+#include <QString>
+#include <QLineEdit>
+#include <QList>
+#include <QSize>
+#include <QResizeEvent>
+class QJSEngine;
+
+namespace Tiled {
+class ScriptDialog : public QDialog
+{
+    Q_OBJECT
+
+public:
+    Q_INVOKABLE ScriptDialog(const QString &title = QString());
+    ~ScriptDialog() override;
+
+    bool checkForClosed() const;
+
+    Q_INVOKABLE void setTitle(const QString &title);
```

Comment:
I don't really see the point of this one, since there is already the `windowTitle` property.